src/Serializer/AbstractItemNormalizer.php

```diff
@@ -222,8 +222,16 @@ public function denormalize(mixed $data, string $class, string $format = null, a
             return $object;
         }
 
+        $options = $this->getFactoryOptions($context);
+        $propertyNames = iterator_to_array($this->propertyNameCollectionFactory->create($resourceClass, $options));
+
         // Revert attributes that aren't allowed to be changed after a post-denormalize check
         foreach (array_keys($data) as $attribute) {
+            $attribute = $this->nameConverter ? $this->nameConverter->denormalize((string) $attribute) : $attribute;
+            if (!\in_array($attribute, $propertyNames, true)) {
+                continue;
+            }
+
             if (!$this->canAccessAttributePostDenormalize($object, $previousObject, $attribute, $context)) {
                 if (null !== $previousObject) {
                     $this->setValue($object, $attribute, $this->propertyAccessor->getValue($previousObject, $attribute));
```

Comment:
There's a change missing line 239 you can reuse the same options.

Reply:
Fixed, thanks :)